### PR TITLE
Only activate+forced_recompute yields encrypted system

### PR DESCRIPTION
**Pull request by okurz**
Also see bsc#993247 and newly created fate#321208

Related progress issue: https://progress.opensuse.org/issues/12782

### CODE REVIEW

**Comment by dzedro** [comment]
There is also https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/tests/installation/grub_test.pm#L38
https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/utils.pm#L321

**Comment by okurz** [comment]
hm, you think I should add

```
-    if (get_var('ENCRYPT') && check_var('ARCH', 'ppc64le')) {
+    if (check_var('ARCH', 'ppc64le') && bootencryptstep_is_applicable) {
```
?

**Comment by dzedro** [comment]
Ignore my comment, it's irrelevant
if disk is encrypted then it should be there and contrarily

File: lib/main_common.pm [hunk 1]
```diff
@@ -207,6 +208,12 @@ sub rescuecdstep_is_applicable {
     return get_var("RESCUECD");
 }
 
+sub bootencryptstep_is_applicable {
+    # activating an existing encrypted volume but not forcing recompute does
+    # *not* yield an encrypted system, see bsc#993247 fate#321208
+    return get_var('ENCRYPT') && !(get_var('ENCRYPT_ACTIVATE_EXISTING' && !get_var('ENCRYPT_FORCE_RECOMPUTE')));
```
Comment (dzedro) [inline lib/main_common.pm]: double negation, what about if I got the point ?
```
return get_var('ENCRYPT') && (!get_var('ENCRYPT_ACTIVATE_EXISTING' && get_var('ENCRYPT_FORCE_RECOMPUTE')));
```

Comment (okurz) [inline lib/main_common.pm]: there was a missing tick `'` behind 'ENCRYPT_ACTIVATE_EXISTING', maybe this confused?

**Comment by okurz** [comment]
The schedule in the four relevant scenarios for **cryptlvm** and derivatives now looks like this. Note the presence or absence of `boot_encrypt`:

## cryptlvm+activate_existing+force_recompute/schedule
isosize tests/installation/isosize.pm
bootloader tests/installation/bootloader.pm
welcome tests/installation/welcome.pm
encrypted_volume_activation tests/installation/encrypted_volume_activation.pm
skip_registration tests/installation/skip_registration.pm
addon_products_sle tests/installation/addon_products_sle.pm
system_role tests/installation/system_role.pm
partitioning tests/installation/partitioning.pm
partitioning_lvm tests/installation/partitioning_lvm.pm
partitioning_finish tests/installation/partitioning_finish.pm
releasenotes tests/installation/releasenotes.pm
installer_timezone tests/installation/installer_timezone.pm
hostname_inst tests/installation/hostname_inst.pm
logpackages tests/installation/logpackages.pm
user_settings tests/installation/user_settings.pm
user_settings_root tests/installation/user_settings_root.pm
installation_overview tests/installation/installation_overview.pm
start_install tests/installation/start_install.pm
install_and_reboot tests/installation/install_and_reboot.pm
**grub_test** tests/installation/grub_test.pm
**boot_encrypt** tests/installation/boot_encrypt.pm
**first_boot** tests/installation/first_boot.pm

## cryptlvm+activate_existing/schedule
isosize tests/installation/isosize.pm
bootloader tests/installation/bootloader.pm
welcome tests/installation/welcome.pm
encrypted_volume_activation tests/installation/encrypted_volume_activation.pm
skip_registration tests/installation/skip_registration.pm
addon_products_sle tests/installation/addon_products_sle.pm
system_role tests/installation/system_role.pm
partitioning tests/installation/partitioning.pm
partitioning_lvm tests/installation/partitioning_lvm.pm
partitioning_finish tests/installation/partitioning_finish.pm
releasenotes tests/installation/releasenotes.pm
installer_timezone tests/installation/installer_timezone.pm
hostname_inst tests/installation/hostname_inst.pm
logpackages tests/installation/logpackages.pm
user_settings tests/installation/user_settings.pm
user_settings_root tests/installation/user_settings_root.pm
installation_overview tests/installation/installation_overview.pm
start_install tests/installation/start_install.pm
install_and_reboot tests/installation/install_and_reboot.pm
**grub_test** tests/installation/grub_test.pm
**first_boot** tests/installation/first_boot.pm

## cryptlvm+cancel_existing/schedule
isosize tests/installation/isosize.pm
bootloader tests/installation/bootloader.pm
welcome tests/installation/welcome.pm
encrypted_volume_activation tests/installation/encrypted_volume_activation.pm
skip_registration tests/installation/skip_registration.pm
addon_products_sle tests/installation/addon_products_sle.pm
system_role tests/installation/system_role.pm
partitioning tests/installation/partitioning.pm
partitioning_lvm tests/installation/partitioning_lvm.pm
partitioning_finish tests/installation/partitioning_finish.pm
releasenotes tests/installation/releasenotes.pm
installer_timezone tests/installation/installer_timezone.pm
hostname_inst tests/installation/hostname_inst.pm
logpackages tests/installation/logpackages.pm
user_settings tests/installation/user_settings.pm
user_settings_root tests/installation/user_settings_root.pm
installation_overview tests/installation/installation_overview.pm
start_install tests/installation/start_install.pm
install_and_reboot tests/installation/install_and_reboot.pm
**grub_test** tests/installation/grub_test.pm
**first_boot** tests/installation/first_boot.pm

## cryptlvm/schedule
isosize tests/installation/isosize.pm
bootloader tests/installation/bootloader.pm
welcome tests/installation/welcome.pm
skip_registration tests/installation/skip_registration.pm
addon_products_sle tests/installation/addon_products_sle.pm
system_role tests/installation/system_role.pm
partitioning tests/installation/partitioning.pm
partitioning_lvm tests/installation/partitioning_lvm.pm
partitioning_finish tests/installation/partitioning_finish.pm
releasenotes tests/installation/releasenotes.pm
installer_timezone tests/installation/installer_timezone.pm
logpackages tests/installation/logpackages.pm
user_settings tests/installation/user_settings.pm
user_settings_root tests/installation/user_settings_root.pm
installation_overview tests/installation/installation_overview.pm
start_install tests/installation/start_install.pm
install_and_reboot tests/installation/install_and_reboot.pm
**grub_test** tests/installation/grub_test.pm
**boot_encrypt** tests/installation/boot_encrypt.pm
**first_boot** tests/installation/first_boot.pm
consoletest_setup tests/console/consoletest_setup.pm
check_console_font tests/console/check_console_font.pm
textinfo tests/console/textinfo.pm
hostname tests/console/hostname.pm
installation_snapshots tests/console/installation_snapshots.pm
snapper_undochange tests/console/snapper_undochange.pm
xorg_vt tests/console/xorg_vt.pm
zypper_lr tests/console/zypper_lr.pm
zypper_ref tests/console/zypper_ref.pm
yast2_lan tests/console/yast2_lan.pm
curl_https tests/console/curl_https.pm
glibc_i686 tests/console/glibc_i686.pm
zypper_in tests/console/zypper_in.pm
yast2_i tests/console/yast2_i.pm
yast2_bootloader tests/console/yast2_bootloader.pm
vim tests/console/vim.pm
firewall_enabled tests/console/firewall_enabled.pm
sshd tests/console/sshd.pm
mtab tests/console/mtab.pm
no_perl_bootloader tests/console/no_perl_bootloader.pm
zypper_lifecycle tests/console/zypper_lifecycle.pm
consoletest_finish tests/console/consoletest_finish.pm
xterm tests/x11/xterm.pm
sshxterm tests/x11/sshxterm.pm
updates_packagekit_gpk tests/update/updates_packagekit_gpk.pm
gnome_control_center tests/x11/gnome_control_center.pm
gnome_terminal tests/x11/gnome_terminal.pm
gedit tests/x11/gedit.pm
firefox tests/x11/firefox.pm
yast2_snapper tests/x11/yast2_snapper.pm
nautilus tests/x11/nautilus.pm
reboot_gnome tests/x11/reboot_gnome.pm
desktop_mainmenu tests/x11/desktop_mainmenu.pm
shutdown tests/x11/shutdown.pm

Schedule was derived by calling `isotovideo` locally on each `vars.json` from a corresponding production run.

File: lib/main_common.pm (hunk 1 above)
Comment (dzedro) [inline lib/main_common.pm]: confusing is  !( !get_var('ENCRYPT_FORCE_RECOMPUTE')

**Comment by dzedro** [comment]
They will not have encrypted partition as we don't "force" encrypted partition
The proposal will - by default - propose what is defined in the control file, which is plain set-up on SLES

**Comment by okurz** [comment]
exactly. Both "activate_existing" and "cancel_existing" will *not* end up with an encrypted system

**Pull request merged**